From e5754399b542640f3f69c5e2513c57a307656032 Mon Sep 17 00:00:00 2001
From: "H.J. Lu" <hjl.tools@gmail.com>
Date: Tue, 5 Aug 2025 09:16:14 -0700
Subject: [PATCH 12/48] Revert "tst-freopen4-main.c: Call
 support_capture_subprocess with chroot"

Revert commit 6463d4a7b28e5ee3891c34a8a1f0a59c24dfa9de to fix

FAIL: stdio-common/tst-freopen4-mem
FAIL: stdio-common/tst-freopen64-4-mem

This fixes BZ #33254.

Reviewed-by: Sam James <sam@gentoo.org>
(cherry picked from commit adec0bf05bc23ec35573c7a5b96440089b69265e)
---
 stdio-common/tst-freopen4-main.c | 44 ++++++++++----------------------
 1 file changed, 14 insertions(+), 30 deletions(-)

diff --git a/stdio-common/tst-freopen4-main.c b/stdio-common/tst-freopen4-main.c
index 436da4d203..3336f5327d 100644
--- a/stdio-common/tst-freopen4-main.c
+++ b/stdio-common/tst-freopen4-main.c
@@ -28,15 +28,25 @@
 #include <support/test-driver.h>
 #include <support/xstdio.h>
 #include <support/xunistd.h>
-#include <support/capture_subprocess.h>
 
-static void
-do_test_chroot (void *data)
+int
+do_test (void)
 {
-  char *temp_dir = (char *) data;
+  mtrace ();
+  char *temp_dir;
   FILE *fp;
   int ret;
 
+  /* These chroot tests verify that either reopening a renamed or
+     deleted file works even in the absence of /proc, or that it fails
+     (without memory leaks); thus, for example, such reopening does
+     not crash in the absence of /proc.  */
+
+  support_become_root ();
+  if (!support_can_chroot ())
+    return EXIT_UNSUPPORTED;
+
+  temp_dir = support_create_temp_directory ("tst-freopen4");
   xchroot (temp_dir);
 
   /* Test freopen with NULL, renamed file.  This verifies that
@@ -86,32 +96,6 @@ do_test_chroot (void *data)
     puts ("freopen of deleted file failed (OK)");
 
   free (temp_dir);
-}
-
-int
-do_test (void)
-{
-  mtrace ();
-  char *temp_dir;
-
-  /* These chroot tests verify that either reopening a renamed or
-     deleted file works even in the absence of /proc, or that it fails
-     (without memory leaks); thus, for example, such reopening does
-     not crash in the absence of /proc.  */
-
-  support_become_root ();
-  if (!support_can_chroot ())
-    return EXIT_UNSUPPORTED;
-
-  temp_dir = support_create_temp_directory ("tst-freopen4");
-
-  struct support_capture_subprocess result;
-  result = support_capture_subprocess (do_test_chroot, temp_dir);
-  support_capture_subprocess_check (&result, "freopen4", 0,
-				    sc_allow_stdout);
-  fputs (result.out.buffer, stdout);
-  support_capture_subprocess_free (&result);
-
   return 0;
 }
 
-- 
2.51.2

